cmd/syncer/main.go

```diff
@@ -33,11 +33,13 @@ import (
 const numThreads = 2
 
 var (
-	fromKubeconfig = flag.String("from_kubeconfig", "", "Kubeconfig file for -from cluster.")
-	fromCluster    = flag.String("from_cluster", "", "Name of the -from logical cluster.")
-	toKubeconfig   = flag.String("to_kubeconfig", "", "Kubeconfig file for -to cluster. If not set, the InCluster configuration will be used.")
-	toContext      = flag.String("to_context", "", "Context to use in the Kubeconfig file for -to cluster, instead of the current context.")
-	clusterID      = flag.String("cluster", "", "ID of the -to cluster. Resources with this ID set in the 'kcp.dev/cluster' label will be synced.")
+	fromKubeconfig         = flag.String("from_kubeconfig", "", "Kubeconfig file for -from cluster.")
+	fromCluster            = flag.String("from_cluster", "", "Name of the -from logical cluster.")
+	toKubeconfig           = flag.String("to_kubeconfig", "", "Kubeconfig file for -to cluster. If not set, the InCluster configuration will be used.")
+	toContext              = flag.String("to_context", "", "Context to use in the Kubeconfig file for -to cluster, instead of the current context.")
+	clusterID              = flag.String("cluster", "", "ID of the -to cluster. Resources with this ID set in the 'kcp.dev/cluster' label will be synced.")
+	leaseControllerEnabled = flag.Bool("lease-controller-enabled", false, "Enable lease controller (should be set to true only in pull mode).")
```

Comment:
The syncer called from `main.go` is always going to be in pull-mode. It's only when `syncer.StartSyncer` etc is wrapped in cluster controller code that it's push mode. So this comment can just drop the parenthetical about pull mode.

Reply:
I wonder if the lease controller needs to be part of the `Syncer` struct or just an additional controller stated in `main`.
Looks like they serve different purposes.

Reply:
That could make sense. Do we think the push-mode syncer will ever use a lease? If not, the lease controller could only be started by `cmd/syncer/main.go`.

Should the lease controller have any hooks into syncer logic? For example, if we fail to renew the lease, should the syncer stop syncing? If those are totally separate responsibilities they make more sense being totally separate in `main.go`, but if they interact with each other they might benefit from being more aware of each other.

Reply:
Good point, i will let it here for now.

I'll see if it makes sense to move it when implementing health check in the cluster controller.